src/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -860,14 +860,34 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_G()
             Assert.Equal(expectedString, result.ToString("g"));
         }
 
-        [Fact]
-        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
+        [Theory, MemberData("Format_String_TestData_O")]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O(DateTimeOffset dt, string expected)
         {
-            DateTimeOffset expected = DateTimeOffset.MaxValue;
-            string expectedString = expected.ToString("o");
+            DateTimeOffset result = DateTimeOffset.ParseExact(dt.ToString("o"), "o", null, DateTimeStyles.None);
+            Assert.Equal(expected, result.ToString("o"));
+        }
+
+        public static IEnumerable<object[]> Format_String_TestData_O()
+        {
+            yield return new object[] { DateTimeOffset.MaxValue, "9999-12-31T23:59:59.9999999+00:00" };
+            yield return new object[] { DateTimeOffset.MinValue, "0001-01-01T00:00:00.0000000+00:00" };
+            yield return new object[] { new DateTimeOffset(1906, 8, 15, 7, 24, 5, 300, new TimeSpan(0, 0, 0)), "1906-08-15T07:24:05.3000000+00:00" };
+            yield return new object[] { new DateTimeOffset(1906, 8, 15, 7, 24, 5, 300, new TimeSpan(7, 30, 0)), "1906-08-15T07:24:05.3000000+07:30" };
```

Comment:
we need to be careful if we add test for values around the saving time transition because this can depend on the current time zone setting on the running machine. 

---
In reply to: [85819453](https://github.com/dotnet/corefx/pull/13180#discussion_r85819453) [](ancestors = 85819453)